ksql-rest-client/src/main/java/io/confluent/ksql/rest/client/KsqlTarget.java

```diff
@@ -171,89 +178,123 @@ private KsqlRequest ksqlRequest(
   }
 
   private <T> RestResponse<T> get(final String path, final Class<T> type) {
-    try (Response response = target
-        .path(path)
-        .request(MediaType.APPLICATION_JSON_TYPE)
-        .headers(headers())
-        .get()
-    ) {
-      return KsqlClientUtil.toRestResponse(response, path, r -> r.readEntity(type));
-    } catch (final Exception e) {
-      throw new KsqlRestClientException("Error issuing GET to KSQL server. path:" + path, e);
-    }
+    return executeRequestSync(HttpMethod.GET, path, null, r -> deserialize(r.getBody(), type));
   }
 
   private <T> RestResponse<T> post(
       final String path,
       final Object jsonEntity,
-      final Optional<Integer> readTimeoutMs,
-      final boolean closeResponse,
-      final Function<Response, T> mapper
+      final Function<ResponseWithBody, T> mapper
   ) {
-    Response response = null;
+    return executeRequestSync(HttpMethod.POST, path, jsonEntity, mapper);
+  }
 
-    try {
-      response = target
-          .path(path)
-          .request(MediaType.APPLICATION_JSON_TYPE)
-          .property(ClientProperties.READ_TIMEOUT, readTimeoutMs.orElse(0))
-          .headers(headers())
-          .post(Entity.json(jsonEntity));
-
-      return KsqlClientUtil.toRestResponse(response, path, mapper);
-    } catch (final ProcessingException e) {
-      if (shouldRetry(readTimeoutMs, e)) {
```

Comment:
Why was this retry logic removed?

Reply:
Aiui, the retry was necessary in the old code because a blocking read was used. This meant the reading the thread was unable to service anything else, like a terminal interrupt. With Vert.x the reads are all non blocking so we don't have this issue.

Reply:
Retries are also useful in case there are transient connectivity issues, though, right? By having retries in the client we shield the user from seeing bizarre, transient failures. Do we still achieve this in the new version?